sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -169,13 +175,19 @@ class QueryExecution(
     // We need to materialize the optimizedPlan here, before tracking the planning phase, to ensure
     // that the optimization time is not counted as part of the planning phase.
     assertOptimized()
-    executePhase(QueryPlanningTracker.PLANNING) {
+    val plan = executePhase(QueryPlanningTracker.PLANNING) {
       // clone the plan to avoid sharing the plan instance between different stages like analyzing,
       // optimizing and planning.
       QueryExecution.prepareForExecution(preparations, sparkPlan.clone())
     }
+    if (!isEager(analyzed)) {
+      tracker.setReadyForExecution(analyzed)
+    }
```

Comment:
if you let `setReadyForExecution` be called multiple times, you can drop the if here.
Add comment
```
// Note: For eagerly executed command it might have already been called in
// `eagerlyExecutedCommand` and is a noop here.
```